src/api/modelTypes.js

```diff
@@ -339,7 +339,7 @@ export type Topic = {|
 // See docs: https://zulip.com/api/construct-narrow
 // prettier-ignore
 /* eslint-disable semi-style */
-export type NarrowElement =
+export type NarrowElement = $ReadOnly<
  | {| +operator: 'is' | 'in' | 'topic' | 'search', +operand: string |}
```

Comment:
This `$ReadOnly` is redundant with the marks on each of the individual properties, where it's spelled `+`. Here's where that syntax is documented:
https://flow.org/en/docs/types/interfaces/#toc-interface-property-variance-read-only-and-write-only

The per-property version is actually the fundamental one -- Flow doesn't have a notion of an object type as a whole being read-only, only of individual properties being that way. When we say a whole type is read-only, it's a shorthand for saying all its properties are. And similarly, `$ReadOnly`'s meaning is defined in terms of making each of the properties read-only:
https://flow.org/en/docs/types/utilities/#toc-readonly

(Similarly at a couple of widget-related types below.)

Reply:
Hmm, thanks! That makes sense.

As it says in the commit message, I intended to make changes only where Flow specifically suggested them. But I don't see an error if I revert this change, and the others I think you've indicated. 🤷

Reply:
This is in `modelTypes`, which is one of the files that it looks like you started with (which makes sense, because these are primarily the types we use to cast the arbitrary JSON we get from the server.)

Reply:
Oh, got it, right!